client/src/Movies/Movie.js

```diff
@@ -5,7 +5,8 @@ const Movie = (props) => {
   const [movie, setMovie] = useState({});
  
   useEffect(() => {
-    const id = 1;
+    //const id = 1;
+    const id = props.movies.find(id => props.match.params.id === `${movie.id}`);
```

Comment:
I think you may be overthinking this line a bit.. You can accomplish setting a dynamic ID with just   `  const id = props.match.params.id;`